app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -11,10 +11,15 @@ class Devise::TwoFactorAuthenticationController < DeviseController
   def new
     current_user.send_new_otp
     flash[:notice] = t('devise.two_factor_authentication.user.new_otp_sent')
-    redirect_to user_two_factor_authentication_path
+    redirect_to user_two_factor_authentication_path(method: 'sms')
```

Comment:
Right, the :new action needs to force SMS OTP screen even for TOTP users.

Maybe I could just add some comments here instead?

Are you suggesting something like redirect_to  user_two_factor_authentication_path_force_sms?
Seems like it would result in the same level of complexity to me.